cmd/cosign/cli/verify/verify_blob.go

```diff
@@ -161,44 +162,69 @@ func VerifyBlobCmd(ctx context.Context, ko sign.KeyOpts, certRef, certEmail, cer
 		if len(uuids) == 0 {
 			return errors.New("could not find a tlog entry for provided blob")
 		}
+		return verifyUUIDs(ctx, ko, rClient, certEmail, certOidcIssuer, sig, b64sig, uuids, blobBytes)
+	}
+
+	// Use the DSSE verifier if the payload is a DSSE with the In-Toto format.
+	if isIntotoDSSE(blobBytes) {
+		verifier = dsse.WrapVerifier(verifier)
+	}
+
+	// verify the signature
+	if err := verifier.VerifySignature(bytes.NewReader([]byte(sig)), bytes.NewReader(blobBytes)); err != nil {
+		return err
+	}
+
+	// verify the rekor entry
+	if err := verifyRekorEntry(ctx, ko, verifier, cert, b64sig, blobBytes); err != nil {
+		return err
+	}
+
+	fmt.Fprintln(os.Stderr, "Verified OK")
+	return nil
+}
 
-		tlogEntry, err := cosign.GetTlogEntry(ctx, rClient, uuids[0])
+func verifyUUIDs(ctx context.Context, ko sign.KeyOpts, rClient *client.Rekor, certEmail, certOidcIssuer, sig, b64sig string, uuids []string, blobBytes []byte) error {
+	var validUUID bool
+	for _, u := range uuids {
+		tlogEntry, err := cosign.GetTlogEntry(ctx, rClient, u)
 		if err != nil {
-			return err
+			continue
 		}
 
 		certs, err := extractCerts(tlogEntry)
 		if err != nil {
-			return err
+			continue
 		}
 
 		co := &cosign.CheckOpts{
 			RootCerts:      fulcio.GetRoots(),
 			CertEmail:      certEmail,
 			CertOidcIssuer: certOidcIssuer,
 		}
-		cert = certs[0]
-		verifier, err = cosign.ValidateAndUnpackCert(cert, co)
+		cert := certs[0]
+		verifier, err := cosign.ValidateAndUnpackCert(cert, co)
 		if err != nil {
-			return err
+			continue
+		}
+		// Use the DSSE verifier if the payload is a DSSE with the In-Toto format.
+		if isIntotoDSSE(blobBytes) {
+			verifier = dsse.WrapVerifier(verifier)
+		}
+		// verify the signature
+		if err := verifier.VerifySignature(bytes.NewReader([]byte(sig)), bytes.NewReader(blobBytes)); err != nil {
+			continue
 		}
-	}
-
-	// Use the DSSE verifier if the payload is a DSSE with the In-Toto format.
-	if isIntotoDSSE(blobBytes) {
-		verifier = dsse.WrapVerifier(verifier)
-	}
 
-	// verify the signature
-	if err := verifier.VerifySignature(bytes.NewReader([]byte(sig)), bytes.NewReader(blobBytes)); err != nil {
-		return err
+		// verify the rekor entry
+		if err := verifyRekorEntry(ctx, ko, verifier, cert, b64sig, blobBytes); err != nil {
+			continue
+		}
+		validUUID = true
```

Comment:
Hmm, looks like we're not doing anything with most of the certs that we check, in which case you probably could stop after the first one.

Longer term I think we want to make sure we print them all or at least record them somehow.

Reply:
Right now all valid entries will be printed in `verifyRekorEntry` which we call for each UUID -- 

https://github.com/sigstore/cosign/blob/ae7f873dd97a60f80a60896c7a2b72c0361a2d4f/cmd/cosign/cli/verify/verify_blob.go#L289

so all valid entries are getting printed to stderr. i can change this to only return one as we were doing before if you think that's better!

sample output:
```
cosign verify-blob --signature sig ./blob
tlog entry verified with uuid: "06b1f4d42d88247a4db98127e47c31ed3cc404024951542c0e98757c965ea7e2" index: 1819736
Verified OK
```

Reply:
Ohhhhhh right, missed that. I was imagining like `cosign verify` where we print out a JSON blob to STDOUT, where you'd want to return a list of all the valid entries for the CLI front-end to handle.

I think keep it for now!